server/store/datastore/migration/004_repos_drop_repo_counter.go

```diff
@@ -0,0 +1,26 @@
+// Copyright 2021 Woodpecker Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package migration
+
+import (
+	"xorm.io/xorm"
+)
+
+var alterTableReposDropCounter = task{
+	name: "alter-table-drop-counter",
+	fn: func(sess *xorm.Session) error {
+		return dropTableColumns(sess, "repos", "repo_counter")
```

Comment:
Don't you need to commit the session after this call? At least there is a warning in the `dropTableColumns` telling you to do so? 🤔

Reply:
do you rever to `// WARNING: YOU MUST COMMIT THE SESSION AT THE END` ?

if you commit the session **within a migration task** -> ALL OTHER alle following migrations are exec on a closed session and will fail!!!

Reply:
tldr: if you commit a session within a migration task, migration is completely broken!!!

Reply:
https://github.com/woodpecker-ci/woodpecker/blob/master/server/store/datastore/migration/migration.go#L89-L91 will do it

Reply:
Ahh, okay so in general the you get the opened session passed to the migration you adjust stuff and at the end the migration logic will commit the session.